src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -137,6 +137,22 @@ protected JestElasticsearchClient(Map<String, String> props, JestClientFactory f
             .preemptiveAuthTargetHosts(address.stream()
                 .map(addr -> HttpHost.create(addr)).collect(Collectors.toSet()));
       }
+
+      final String proxy =
+              config.getString(ElasticsearchSinkConnectorConfig.CONNECTION_PROXY_CONFIG);
+      if (proxy != null && !proxy.isEmpty()) {
```

Comment:
this looks certainly good to me, might be wrap this into their own function? this method starts to be quite big, right?

Reply:
Sure. Moved proxy configuration to a separate method.